src/praisonai/praisonai/scheduler/condition_gate.py

```
@@ -0,0 +1,103 @@
+
+    def __init__(self, *, timeout: float = 30.0) -> None:
+        self._timeout = timeout
+
+    def should_run(self, job: Any) -> GateResult:
+        command = (getattr(job, "pre_run", None) or "").strip()
+        if not command:
+            # No gate configured → always run (unconditional, as today).
+            return GateResult(run=True)
+
+        try:
+            completed = subprocess.run(
```

Ruff (0.15.18)

[error] `subprocess` call with `shell=True` identified, security issue

(S602)